managers/backup-manager/BackupStatusPoller.js

```diff
@@ -0,0 +1,277 @@
+    assert.ok(_.get(opts, 'backup_guid'), `Argument 'opts.backup_guid' is required to start polling for backup`);
+    assert.ok(_.get(opts, 'plan_id'), `Argument 'opts.plan_id' is required to start polling for backup`);
+    assert.ok(_.get(opts, 'started_at'), `Argument 'opts.started_at' is required to start polling for backup`);
+    assert.ok(_.get(opts, 'deployment'), `Argument 'opts.deployment' is required to start polling for backup`);
+
+    logger.info('Checking Operation Completion Status for :', opts);
+    const operationName = CONST.OPERATION_TYPE.BACKUP;
+    const instance_guid = opts.instance_guid;
+    const backup_guid = opts.backup_guid;
+    const plan = catalog.getPlan(opts.plan_id);
+    const deployment = opts.deployment;
+
+    return BackupService.createService(plan)
+      .then(backupService => backupService.getOperationState(CONST.OPERATION_TYPE.BACKUP, opts))
+      .then(operationStatusResponse => {
+        if (utils.isServiceFabrikOperationFinished(operationStatusResponse.state)) {
+          return operationStatusResponse;
+        } else {
+          logger.info(`Instance ${instance_guid} ${operationName} for backup guid ${backup_guid} still in-progress - `, operationStatusResponse);
+          const currentTime = new Date();
+          const backupTriggeredDuration = (currentTime - new Date(opts.started_at)) / 1000;
+          return Promise
+            .try(() => eventmesh
+              .apiServerClient
+              .patchResource({
+                resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+                resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                resourceId: opts.backup_guid,
+                status: {
+                  response: operationStatusResponse
+                }
+              }))
+            .then(() => {
+              const lockDeploymentMaxDuration = eventmesh.lockManager.getLockTTL(operationName);
+              if (backupTriggeredDuration > lockDeploymentMaxDuration) {
+                //Operation timed out
+                if (!opts.abortStartTime) {
+                  //Operation not aborted. Aborting operation and with abort start time
+                  let abortStartTime = new Date().toISOString();
+                  return eventmesh.apiServerClient.patchResource({
+                      resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.BACKUP,
+                      resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
+                      resourceId: opts.backup_guid,
+                      status: {
+                        state: CONST.APISERVER.RESOURCE_STATE.ABORT,
+                        response: {
+                          abortStartTime: abortStartTime
+                        }
+                      }
+                    })
+                    .then(() => {
+                      operationStatusResponse.state = CONST.OPERATION.ABORTING;
```

Comment:
where the actual abort happening by calling agent?

Reply:
[Here](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/411/files#diff-3ab340ff5bc3b2adabd32dd87a7c77afR68) when state is changed to `ABORT`, backup-manager picks it up and aborts.